### DISCUSSION TITLE
Forced tunneling via S2S VPN

### DISCUSSION THREAD

**Discussion by rabusiak** [AzureCommunityPL/group#381] [Pomoc]
Czołem,
Mam taką potrzebę, żeby ruch do internetu z konkretnej vmki w Azure był tunelowany przez firewalla w HQ, który ma stałe ip whitelistowane w wielu, wielu miejscach ;) Mam już tunel Azure - firewall. Wszystko cacy, komunikacja tryka, produkcja jednym słowem :) Znalazłem taki o to [docs](https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-forced-tunneling-rm), który mniej więcej robi to co potrzebuję - sobie tę vmkę wydzielę do osobnego subnetu.
![1](https://user-images.githubusercontent.com/61978743/147818570-8468e43b-ef7d-40d7-b34b-82cf0eeaba39.jpg)

Mam pytanie, czy ktoś już taką konfigurację testował?

Szczególnie wątpliwości mam w kroku 7:
"Assign a default site to the virtual network gateway. The -GatewayDefaultSite is the cmdlet parameter that allows the forced routing configuration to work, so take care to configure this setting properly."
`Set-AzVirtualNetworkGatewayDefaultSite -GatewayDefaultSite $LocalGateway -VirtualNetworkGateway $VirtualGateway`

Ustawienie defaultowego site'a nie wpłynie na działanie subnetów bez nowo-utworzonego UDR'a?
Nie chciałbym przerutować przez przypadek całego ruchu do internetu przez fw w HQ ze wszystkich subnetów. Zabiłbym go :P

**Comment by senionit** [comment]
Cześć,

1. Od razu mówię, że nie testowałem.
2. Jak dla mnie to musisz rozpatrzyć kilka elementów:
    a. Tablica routingu jest przed VPNem. Co oznacza, że jeżeli masz standardowy default gateway w innych subnetach to poleci nim a nie do VPG.
   b. Przykład pokazuje vpn-a bez BGP, więc pytanie, czy masz BGP bo ono może wstrzyknąć default route i wtedy może być płacz i zgrzytanie zębów.
3. Jak dla mnie sam kontrowersyjny krok mówi tylko o przygotowaniu drugiej strony do przyjęcia pełnego ruchu z azure (pod warunkiem, że ruch zostanie tam wysłany)
     * https://docs.microsoft.com/en-us/powershell/module/az.network/set-azvirtualnetworkgatewaydefaultsite?view=azps-7.0.0


pozdrawiam
Sebastian
